server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java

```diff
@@ -70,6 +72,37 @@
   // createSecret
   //---------------------------------------------------------------------------------------
 
+  @Test public void renamesSecret() throws Exception {
+    String oldName = UUID.randomUUID().toString();
+    createSecret(oldName);
+
+    String newName = UUID.randomUUID().toString();
+    assertEquals(200, renameSecret(oldName, newName));
+
+    assertFalse(getSecret(oldName).isPresent());
+    assertTrue(getSecret(newName).isPresent());
+  }
+
+  @Test public void cannotRenameSecretToExistingSecret() throws Exception {
+    String oldName = UUID.randomUUID().toString();
+    createSecret(oldName);
+
+    String newName = UUID.randomUUID().toString();
+    createSecret(newName);
+
+    assertEquals(500, renameSecret(oldName, newName));
+  }
```

Comment:
Upon further reflection, I think this error code is inconsistent with current behavior. Trying to create the same secret twice results in a ConflictException (error code 409), and I believe we should be doing the same here.